tensorrt_llm/_torch/modules/linear.py

```
@@ -1382,13 +1382,24 @@ def _input_prepare(self, module: Linear, input: torch.Tensor):
     def apply(self, module: Linear, input: torch.Tensor,
               bias: Optional[torch.Tensor]):
         # Handle multi-dimensional inputs (e.g., 3D: batch, seq, hidden).
-        # GEMM requires 2D. Only plain tensors support for now, skip for
-        # tuple and Fp4QuantizedTensor.
+        # NVFP4 GEMM requires a 2D mat1; flatten here and unflatten the output below.
         original_shape = None
         if not isinstance(input,
                           (tuple, Fp4QuantizedTensor)) and input.dim() > 2:
             original_shape = input.shape
             input = input.reshape(-1, input.shape[-1])
```

Ruff (0.15.15)

[error] Variable `input` is shadowing a Python builtin

(A001)